### PR TITLE
Add test for Manager.share and update dat.download

### PR DESCRIPTION
**Pull request by joehand**
* Adds test for `Manager.share` function
* Updates `dat.download` to use pump (was exiting too early before)
* Add pump + through2 to dependencies

### CODE REVIEW

File: dat.js [hunk 1]
```diff
@@ -90,8 +94,9 @@ Dat.prototype.download = function (link, location, cb) {
       debug('done', entry.size)
     })
   })
-  metadata.on('error', cb)
-  metadata.on('end', function () {
```
Comment (joehand) [inline dat.js]: This was causing download to callback too early (and I think why share sometimes didn't work). The `end` event on metadata can fire before downloads are done.

File: tests/test.js [hunk 2]
```diff
@@ -1,6 +1,8 @@
 var Manager = require('../')
 var Dat = require('dat')
+var LocalDat = require('../dat')
```
Comment (joehand) [inline tests/test.js]: A bit awkward...

I wanted to test using the local one to make sure it worked on both ends too.

Comment (karissa) [inline tests/test.js]: Ahh makes sense

File: dat.js (hunk 1 above)
Comment (karissa) [inline dat.js]: Ahhh ok!

**Comment by karissa** [comment]
Thanks for doing this. Are you finished?

**Comment by joehand** [comment]
Oh, yep! All done. Thanks.

**Comment by karissa** [comment]
Thank you!

**Pull request merged**